### PR TITLE
Add filter for non-synonymous mutations and re-run TMB with bug fix

### PR DESCRIPTION
**Pull request by cansavvy**
# Purpose/implementation Section

#### What scientific question is your analysis addressing?
- Adding an option to drop the synonymous mutations from the TMB calculations. 
- Re-running downstream TMB items after getting them updated. 

I would like to do some more in depth comparison of the old and new results, but I will save that for a different PR so as to not have too many things at once. 

#### What was your approach?
- Added a filtering option `--nonsynfilter` which is a TRUE/FALSE on whether mutations should be filtered to be only non-synonymous mutations. 
- I re-ran both TCGA and PBTA TMBs
- I re-ran the tmb-compare-notebook and adjusted some of the margins so it would be better aligned. 
- I re-ran the `fig2-mutational-landscape.R` figure again so it's updated. 

#### What GitHub issue does your pull request address?
#726 

### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

- Is there a way I can make the margins adjust more automatically? It was quite a bit of finagling of margins and text size to get the plots to be aligned. Perhaps this is just how it has to be. 
- Anything I'm missing? 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes. The TMBs for the adults do seem slightly higher? That's good. 

#### What is your summary of the results?
Still figuring this out. It's not changed as I suspected it would be. 
*Note that data files themselves are not pushed to GitHub but we will need to update them in the next data release. 

#### Reproducibility Checklist
- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist
- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by cansavvy** [comment]
This is a reminder to myself to update the blog post with these updates to TMB. @cansavvy

File: analyses/snv-callers/scripts/03-calculate_tmb.R [hunk 1]
```diff
@@ -146,6 +154,19 @@ join_cols <- c(
...
+  "Missense_Mutation",
+  "Frame_Shift_Del",
+  "In_Frame_Ins",
+  "Frame_Shift_Ins",
+  "Splice_Site",
+  "Nonsense_Mutation",
+  "In_Frame_Del",
+  "Nonstop_Mutation",
+  "Translation_Start_Site"
+)
```
Comment (jashapiro) [inline analyses/snv-callers/scripts/03-calculate_tmb.R:168]: As we have discussed, this does differ slightly from the standard proposed in https://jitc.bmj.com/content/8/1/e000147#DC by the *Friends of Cancer Research TMB Harmonization Project*. 

The relevant table is here: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/725#issuecomment-657575733

In particular, the FoCR group does not include `Translation_Start_Site`, `Nonstop_Mutation` or `Splice_Site`. I tend to think all of those are worth including, but I guess one simple question is how common these mutations actually are in the data set. Either way, we will want to standardize our criteria across analyses.

**Comment by cansavvy** [comment]
> This looks great. Do we want to include a version without the nonsyn filter as well, just for comparison? I would not put that in the final figure, but having the figure and/or data table might be nice.

I want to do this, but figured I should make it a separate PR so this one stays focused.

File: analyses/snv-callers/scripts/03-calculate_tmb.R (hunk 1 above)
Comment (cansavvy) [reply analyses/snv-callers/scripts/03-calculate_tmb.R:168]: I think this question is something I can ask in a next PR that does some more exploration. I'll file an issue that lays out what kinds of exploration analyses I think should be included.

Comment (cansavvy) [reply analyses/snv-callers/scripts/03-calculate_tmb.R:168]: Filed the issue: https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/729

**Comment by cansavvy** [comment]
The follow up questions, I'll address on separate PRs, notoed on this issue here: https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/729

For wrapping up this current PR, I'll play around a bit with finding a better way to "automagically" align the tmb plots.

**Comment by cansavvy** [comment]
I tried `patchwork`. But am encountering an error like what's documented here: https://github.com/thomasp85/patchwork/issues/148
I tried minimizing the `theme` arguments specified but it still didn't fix the problem.

**Comment by cansavvy** [comment]
@jashapiro I actually use `cowplot` in the `tmb-compare-tcga` notebook, but for whatever reason the `align = "v"` argument doesn't work. Let me know if you have insight into why its broken/being ignored by cowplot: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/tmb-compare-tcga/compare-tmb.Rmd#L141

**Comment by cansavvy** [comment]
Not sure what else to try for the "automagic" y axis alignment (see above comments). But in the interest of moving this and the future issues (#729) forward, I've wrote this problem down here: https://github.com/cansavvy/openpbta-notebook-concept/issues/9 and will return to it if we have any breakthroughs. 

With this being noted, are we ready to merge, @jashapiro ?

**Comment by jashapiro** [comment]
> With this being noted, are we ready to merge, @jashapiro ?
:shipit:

**Pull request merged**